kinesumer.go

```diff
@@ -132,6 +132,30 @@ func (k *Kinesumer) GetShards() (shards []*kinesis.Shard, err error) {
 	return
 }
 
+func (k *Kinesumer) LaunchShardWorker(shards []*kinesis.Shard) (int, error) {
+	perm := rand.Perm(len(shards))
+	for _, j := range perm {
+		err := k.StateSync.TryAcquire(shards[j].ShardID)
+		if err == nil {
+			worker := &ShardWorker{
+				kinesis:         k.Kinesis,
+				shard:           shards[j],
+				stateSync:       k.StateSync,
+				stream:          k.Stream,
+				pollTime:        k.opt.PollTime,
+				stop:            k.stop,
+				stopped:         k.stopped,
+				c:               k.records,
+				GetRecordsLimit: k.opt.GetRecordsLimit,
+			}
+			go worker.RunWorker()
+			k.nRunning++
+			return j, nil
+		}
+	}
+	return 0, errors.New("Could not launch worker")
```

Comment:
Hmm, this is going to fail when the process dies but comes back up within the ttl. You'll probably need to retry for longer than the ttl before you are sure that all shards are taken.

What are some of your tests cases? Please add tests